sos/plugins/nvme.py

```diff
@@ -1,119 +1,98 @@
-            pci = subprocess.check_output("lscfg -vl %s | grep nvme | grep %s | awk '{ print $1 }'" 
-                % (dev[0:-2], grep_op), shell=True).strip()
-
-            # add slot and pci location in sos_strings/nvme/
-            self.add_string_as_file(slot, "slot_location.%s" % dev)
-            self.add_string_as_file(pci, "pci_location.%s" % dev)
-
-            # get PCI Vendor ID
-            self.add_cmd_output("sh -c \"lspci -vs %s | grep 'Non-Volatile memory controller' | cut -c 46-\"" 
-                % pci, suggest_filename="pci_vdid.%s" % dev)
-
-            # get Subsystem ID
-            self.add_cmd_output("sh -c \"lspci -vs %s | grep Subsystem | cut -c 13-\"" 
-                % pci, suggest_filename="pci_ssid.%s" % dev)
-
-            # get Part Number
-            self.add_cmd_output("sh -c \"lspci -vvs %s | grep '\[PN\]' | awk '{ print $4 }'\"" 
-                % pci, suggest_filename="part-number.%s" % dev)
-
-            # get Engineering Changes [EC] code
-            self.add_cmd_output("sh -c \"lspci -vvs %s | grep '\[EC\]' | awk '{ print $4 }'\"" 
-                % pci, suggest_filename="engineering-changes.%s" % dev)
-
-            # check if nvme-cli package is installed
-            if self.is_installed("nvme-cli"):
-                self.add_cmd_output("nvme list")
-                self.add_cmd_output("nvme list-ns /dev/%s" % dev, suggest_filename="list-ns.%s" % dev)
-                self.add_cmd_output("nvme fw-log /dev/%s" % dev, suggest_filename="fw-log.%s" % dev)
-                self.add_cmd_output("nvme list-ctrl /dev/%s" % dev, suggest_filename="list-ctrl.%s" % dev)
-                self.add_cmd_output("nvme id-ctrl -H /dev/%s" % dev, suggest_filename="id-ctrl.%s" % dev)
-                self.add_cmd_output("nvme id-ns -H /dev/%s" % dev, suggest_filename="id-ns.%s" % dev)
-                self.add_cmd_output("nvme smart-log /dev/%s" % dev, suggest_filename="smart-log.%s" % dev)
-                self.add_cmd_output("nvme error-log /dev/%s" % dev, suggest_filename="error-log.%s" % dev)
-
-            else:
-                self.add_string_as_file(("The nvme-cli tool is not installed. If you want more configuration details and" 
-                    " system information about NVMe devices, you should install it.\n"), "nvme-cli_not_installed")	
+            cmd_lscfg = self.call_ext_prog("lscfg -vl %s" % dev[0:-2])
+            if cmd_lscfg['status'] == 0:
+                pci_slot_location = self.get_pci_slot_location(
+                                cmd_lscfg['output'], op)
+
+                if pci_slot_location:
+                    pci_loc = pci_slot_location[0]
+                    slot_loc = pci_slot_location[3]
+                    self.add_string_as_file(pci_loc, "pci_loc.%s" % dev)
+                    self.add_string_as_file(slot_loc, "slot_loc.%s" % dev)
+
+            # runs nvme-cli commands
+            self.add_cmd_output([
+                                "nvme list",
```

Comment:
These lines should only be indented *one* level from the preceding line (same as any other Python block indent).